modules/renter.go

```diff
@@ -56,46 +75,64 @@ type FileInfo struct {
 	Expiration     types.BlockHeight `json:"expiration"`
 }
 
-// DownloadInfo provides information about a file that has been requested for
-// download.
-type DownloadInfo struct {
-	SiaPath     string    `json:"siapath"`
-	Destination string    `json:"destination"`
-	Filesize    uint64    `json:"filesize"`
-	Received    uint64    `json:"received"`
-	StartTime   time.Time `json:"starttime"`
-}
-
-// An Allowance dictates how much the Renter is allowed to spend in a given
-// period. Note that funds are spent on both storage and bandwidth.
-type Allowance struct {
-	Funds       types.Currency    `json:"funds"`
-	Hosts       uint64            `json:"hosts"`
-	Period      types.BlockHeight `json:"period"`
-	RenewWindow types.BlockHeight `json:"renewwindow"`
-}
-
-// RenterSettings control the behavior of the Renter.
-type RenterSettings struct {
-	Allowance Allowance `json:"allowance"`
-}
-
 // A HostDBEntry represents one host entry in the Renter's host DB. It
 // aggregates the host's external settings and metrics with its public key.
+//
+// TODO: WE FORGOT THE JSON STRINGS FOR SOME OF THE ELEMENTS! It's used in both
+// persist and over the API.
 type HostDBEntry struct {
 	HostExternalSettings
 	PublicKey types.SiaPublicKey `json:"publickey"`
 	// ScanHistory is the set of scans performed on the host. It should always
 	// be ordered according to the scan's Timestamp, oldest to newest.
-	ScanHistory HostDBScans
+	ScanHistory HostDBScans `json:"scanhistory"`
 	// FirstSeen is the last block height at which this host was announced.
-	FirstSeen types.BlockHeight
+	FirstSeen types.BlockHeight `json:"firstseen"`
 }
 
 // HostDBScan represents a single scan event.
 type HostDBScan struct {
-	Timestamp time.Time
-	Success   bool
+	Timestamp time.Time `json:"timestamp"`
+	Success   bool      `json:"success"`
+}
+
+// HostScoreBreakdown provides a piece-by-piece explanation of why a host has
+// the score that they do.
+//
+// NOTE: Renters are free to use whatever scoring they feel appropriate for
+// hosts. Some renters will outright blacklist or whitelist sets of hosts. The
+// results provided by this struct can only be used as a guide, and may vary
+// significantly from machine to machine.
+type HostScoreBreakdown struct {
+	AgeAdjustment              float64 `json:"ageadjustment"`
+	BurnAdjustment             float64 `json:"burnadjustment"`
+	CollateralAdjustment       float64 `json:"collateraladjustment"`
+	PriceAdjustment            float64 `json:"pricesmultiplier"`
+	StorageRemainingAdjustment float64 `json:"storageremainingadjustment"`
+	UptimeAdjustment           float64 `json:"uptimeadjustment"`
+	VersionAdjustment          float64 `json:"versionadjustment"`
+}
```

Comment:
I decided that everything in this struct would either be a float or a bool. I could have added a final score, but decided not to. If we ever deprecate a float, it'll always return '1'. If we deprecate a bool, it'll always return 'false'.

The bools we have time to change our mind about. But basically they'd be things like 'IsEurope' or 'IsHIPPACertified', stuff like that. If we deprecate a category, nothing is in that category anymore. Generally, a bool does not affect your score except to disqualify you from a bucket. For example, we may require that there be at least 2 hosts on every continent.

That's how I decided to handle compatibility here.